### ISSUE TITLE
mypy does not recognize keyword arguments to json_serializable_dataclass instances

### ISSUE THREAD

**Issue by kevinsung** [quantumlib/Cirq#2769]
**To reproduce:**

1.  Create a file `test.py` in `cirq/` with the following contents:
```
import dataclasses
from cirq import protocols


@dataclasses.dataclass
class A:
    a: int


@protocols.json_serializable_dataclass
class B:
    b: int


A(a=0)
B(b=0)
```
2. Execute `check/mypy`. This gives the following output:
> cirq/test.py:16: error: Unexpected keyword argument "b" for "B"

**Comment by mpharrigan** [comment]
looks fairly hopeless: https://mypy.readthedocs.io/en/stable/additional_features.html#caveats-known-issues

**Comment by dabacon** [comment]
Do we want to remove json_serializable_dataclass decorator?  It doesn't look like ti is used anywhere.

**Comment by kevinsung** [comment]
It's used once in `experiments/cross_entropy_benchmarking.py` but that shouldn't prevent us from removing it.

**Comment by mpharrigan** [comment]
I would not object to removing it, if the drawbacks exceed the marginal benefit

**Comment by 95-martin-orion** [comment]
Resolved by #5208.
